cli/src/commands/edit.rs

```diff
@@ -21,10 +21,11 @@ use crate::cli_util::{CommandHelper, RevisionArg};
 use crate::command_error::CommandError;
 use crate::ui::Ui;
 
-/// Edit a commit in the working copy
+/// Sets the specified revision as the working-copy revision
 ///
-/// Puts the contents of a commit in the working copy for editing. Any changes
-/// you make in the working copy will update (amend) the commit.
+/// Note: it is generally recommended to instead to use `jj new` and `jj
+/// squash`. Creating a new revision enables being more intentional with respect
+/// to the existing revision.
```

Comment:
Yeah, I can see how it's poor here. There's a mindset associated with the phrasing that I'd love to talk about, probably will come up if/when we talk governance/process.

I think it might be best to drop the line and have people get the nuance from the FAQ.